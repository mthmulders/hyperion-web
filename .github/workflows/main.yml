name: Main workflow

on:
  push:
    branches: [ master, mutationtesting ]
  pull_request:
    branches: [ master ]
    types: [opened, synchronize, reopened]

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout code
        uses: actions/checkout@v2
        with:
          # Disabling shallow clone is recommended for improving relevancy of reporting
          fetch-depth: 0

      - name: Setup Node.js environment
        uses: actions/setup-node@v2.1.3
        with:
          node-version: 12.18.3

      - name: Set up cache for node_modules
        uses: actions/cache@v2.1.3
        with:
          path: "**/node_modules"
          key: hyperion-web-${{ hashFiles('package.json') }}-${{ hashFiles('yarn.lock') }}
          restore-keys: |
            hyperion-web
      
      - name: Download Node modules
        run: yarn install

      - name: Build and test code
        run: npm run test:ci
        env:
          CI: true
          JEST_JUNIT_OUTPUT_DIR: test-reports
      
      - name: Create production build
        run: |
          npm run build
          pushd build
          zip -9vr ./hyperion-web.zip .
          popd
      
      - name: Upload application
        uses: actions/upload-artifact@v2
        if: github.ref == 'refs/heads/master'
        with:
          name: hyperion-web.zip
          path: build/hyperion-web.zip

  mutationtesting:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout code
        uses: actions/checkout@v2
        with:
          # Disabling shallow clone is recommended for improving relevancy of reporting
          fetch-depth: 0

      - name: Setup Node.js environment
        uses: actions/setup-node@v2.1.3
        with:
          node-version: 12.18.3

      - name: Set up cache for node_modules
        uses: actions/cache@v2.1.3
        with:
          path: "**/node_modules"
          key: hyperion-web-${{ hashFiles('package.json') }}-${{ hashFiles('yarn.lock') }}
          restore-keys: |
            hyperion-web

      - name: Download Node modules
        run: yarn install

      - name: Run Stryker
        run: npm run stryker
        env:
          STRYKER_DASHBOARD_API_KEY: ${{ secrets.STRYKER_DASHBOARD_API_KEY }}
