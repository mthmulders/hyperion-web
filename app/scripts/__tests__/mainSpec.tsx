import * as React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Main } from '../main';

describe('<Main />', () => {
    it('should render application components', () => {
        // Act
        const main = shallow(<Main />);

        // Assert
        expect(main).toMatchSnapshot();
    });
});