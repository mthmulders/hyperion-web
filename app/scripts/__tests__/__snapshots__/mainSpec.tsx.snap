exports[`<Main /> should render application components 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }>
    <div>
        <AppBar
            onLeftIconButtonTouchTap={[Function]}
            showMenuIconButton={true}
            title="Υπερίων"
            zDepth={1} />
        <Drawer
            disableSwipeToOpen={false}
            docked={false}
            onRequestChange={[Function]}
            open={false}
            openSecondary={false}
            swipeAreaWidth={30}
            width={null}
            zDepth={2} />
        <Router
            history={
                Object {
                    "createHref": [Function],
                    "createKey": [Function],
                    "createLocation": [Function],
                    "createPath": [Function],
                    "getCurrentLocation": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "listen": [Function],
                    "listenBefore": [Function],
                    "push": [Function],
                    "replace": [Function],
                    "transitionTo": [Function],
                    "unsubscribe": [Function],
                  }
            }
            render={[Function]}>
            <Route
                component={[Function]}
                path="/" />
        </Router>
    </div>
</Provider>,
  "nodes": Array [
    <Provider
      store={
            Object {
                  "dispatch": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                  Symbol(observable): [Function],
                }
      }>
      <div>
            <AppBar
                  onLeftIconButtonTouchTap={[Function]}
                  showMenuIconButton={true}
                  title="Υπερίων"
                  zDepth={1} />
            <Drawer
                  disableSwipeToOpen={false}
                  docked={false}
                  onRequestChange={[Function]}
                  open={false}
                  openSecondary={false}
                  swipeAreaWidth={30}
                  width={null}
                  zDepth={2} />
            <Router
                  history={
                        Object {
                              "createHref": [Function],
                              "createKey": [Function],
                              "createLocation": [Function],
                              "createPath": [Function],
                              "getCurrentLocation": [Function],
                              "go": [Function],
                              "goBack": [Function],
                              "goForward": [Function],
                              "listen": [Function],
                              "listenBefore": [Function],
                              "push": [Function],
                              "replace": [Function],
                              "transitionTo": [Function],
                              "unsubscribe": [Function],
                            }
                  }
                  render={[Function]}>
                  <Route
                        component={[Function]}
                        path="/" />
            </Router>
      </div>
</Provider>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Main />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Main {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "open": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }>
          <div>
                    <AppBar
                              onLeftIconButtonTouchTap={[Function]}
                              showMenuIconButton={true}
                              title="Υπερίων"
                              zDepth={1} />
                    <Drawer
                              disableSwipeToOpen={false}
                              docked={false}
                              onRequestChange={[Function]}
                              open={false}
                              openSecondary={false}
                              swipeAreaWidth={30}
                              width={null}
                              zDepth={2} />
                    <Router
                              history={
                                        Object {
                                                  "createHref": [Function],
                                                  "createKey": [Function],
                                                  "createLocation": [Function],
                                                  "createPath": [Function],
                                                  "getCurrentLocation": [Function],
                                                  "go": [Function],
                                                  "goBack": [Function],
                                                  "goForward": [Function],
                                                  "listen": [Function],
                                                  "listenBefore": [Function],
                                                  "push": [Function],
                                                  "replace": [Function],
                                                  "transitionTo": [Function],
                                                  "unsubscribe": [Function],
                                                }
                              }
                              render={[Function]}>
                              <Route
                                        component={[Function]}
                                        path="/" />
                    </Router>
          </div>
</Provider>,
        "_debugID": 2,
        "_renderedOutput": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }>
          <div>
                    <AppBar
                              onLeftIconButtonTouchTap={[Function]}
                              showMenuIconButton={true}
                              title="Υπερίων"
                              zDepth={1} />
                    <Drawer
                              disableSwipeToOpen={false}
                              docked={false}
                              onRequestChange={[Function]}
                              open={false}
                              openSecondary={false}
                              swipeAreaWidth={30}
                              width={null}
                              zDepth={2} />
                    <Router
                              history={
                                        Object {
                                                  "createHref": [Function],
                                                  "createKey": [Function],
                                                  "createLocation": [Function],
                                                  "createPath": [Function],
                                                  "getCurrentLocation": [Function],
                                                  "go": [Function],
                                                  "goBack": [Function],
                                                  "goForward": [Function],
                                                  "listen": [Function],
                                                  "listenBefore": [Function],
                                                  "push": [Function],
                                                  "replace": [Function],
                                                  "transitionTo": [Function],
                                                  "unsubscribe": [Function],
                                                }
                              }
                              render={[Function]}>
                              <Route
                                        component={[Function]}
                                        path="/" />
                    </Router>
          </div>
</Provider>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Main />,
}
`;
